main.go

```diff
@@ -38,8 +42,13 @@ func healthCheckHandler(w http.ResponseWriter, r *http.Request) {
 func corsMiddleware(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		origin := r.Header.Get("Origin")
```

Comment:
Testing Advice: This middleware logic (and the auth logic below) is critical. I recommend adding unit tests using the `net/http/httptest` package to verify:
1.  CORS headers are set correctly for allowed origins.
2.  Preflight (OPTIONS) requests are handled.
3.  Unauthorized requests are rejected (when auth is enabled).